# AirBnB NY Locations Data Case Study

In this final project, you task will be to take the data provided and find evidance to answer the following questions.

1. Which hosts are the busiest and why?
2. How many neighborhood groups are available and which shows up the most?
3. Are private rooms the most popular in manhattan?
4. Which hosts are the busiest and based on their reviews?
5. Which neighorhood group has the highest average price?
6. Which neighborhood group has the highest total price?
7. Which top 5 hosts have the highest total price?
8. Who currently has no (zero) availability with a review count of 100 or more?
9. What host has the highest total of prices and where are they located?
10. When did Danielle from Queens last receive a review?

You will be given **4 hours** to complete this assignment. 
**Be Advised** I will go dark for this intire assignment time period. That said, any questions that you would like to ask about the data, or the project **MUST** be asked before the time starts. Once the time has started, I can no longer give information.

This is to similate what you will face when you are out in the wild. 

Happy Coding!

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [4]:
air_bnb = pd.read_csv('./AB_NYC_2019.csv')
air_bnb

id                                               name   host_id  \
0          2539                 Clean & quiet apt home by the park      2787   
1          2595                              Skylit Midtown Castle      2845   
2          3647                THE VILLAGE OF HARLEM....NEW YORK !      4632   
3          3831                    Cozy Entire Floor of Brownstone      4869   
4          5022   Entire Apt: Spacious Studio/Loft by central park      7192   
...         ...                                                ...       ...   
48890  36484665    Charming one bedroom - newly renovated rowhouse   8232441   
48891  36485057      Affordable room in Bushwick/East Williamsburg   6570630   
48892  36485431            Sunny Studio at Historical Neighborhood  23492952   
48893  36485609               43rd St. Time Square-cozy single bed  30985759   
48894  36487245  Trendy duplex in the very heart of Hell's Kitchen  68119814   

           host_name neighbourhood_group       neighbourhood  latitude  \
0               John            Brooklyn          Kensington  40.64749   
1           Jennifer           Manhattan             Midtown  40.75362   
2          Elisabeth           Manhattan              Harlem  40.80902   
3        LisaRoxanne            Brooklyn        Clinton Hill  40.68514   
4              Laura           Manhattan         East Harlem  40.79851   
...              ...                 ...                 ...       ...   
48890        Sabrina            Brooklyn  Bedford-Stuyvesant  40.67853   
48891        Marisol            Brooklyn            Bushwick  40.70184   
48892  Ilgar & Aysel           Manhattan              Harlem  40.81475   
48893            Taz           Manhattan      Hell's Kitchen  40.75751   
48894     Christophe           Manhattan      Hell's Kitchen  40.76404   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
0      -73.97237     Private room    149               1                  9   
1      -73.98377  Entire home/apt    225               1                 45   
2      -73.94190     Private room    150               3                  0   
3      -73.95976  Entire home/apt     89               1                270   
4      -73.94399  Entire home/apt     80              10                  9   
...          ...              ...    ...             ...                ...   
48890  -73.94995     Private room     70               2                  0   
48891  -73.93317     Private room     40               4                  0   
48892  -73.94867  Entire home/apt    115              10                  0   
48893  -73.99112      Shared room     55               1                  0   
48894  -73.98933     Private room     90               7                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
0      2018-10-19               0.21                               6   
1      2019-05-21               0.38                               2   
2             NaN                NaN                               1   
3      2019-07-05               4.64                               1   
4      2018-11-19               0.10                               1   
...           ...                ...                             ...   
48890         NaN                NaN                               2   
48891         NaN                NaN                               2   
48892         NaN                NaN                               1   
48893         NaN                NaN                               6   
48894         NaN                NaN                               1   

       availability_365  
0                   365  
1                   355  
2                   365  
3                   194  
4                     0  
...                 ...  
48890                 9  
48891                36  
48892                27  
48893                 2  
48894                23  

[48895 rows x 16 columns]

In [11]:
# How many neighborhood groups are available?
print(f'Total Neighborhood Groups: {len(air_bnb.neighbourhood_group.unique())}')

# Which shows up the most?
boroughs = air_bnb.groupby('neighbourhood_group')['id'].count()
print(boroughs)

print('Manhattan is the neighborbood that shows up the most.')


Total Neighborhood Groups: 5
neighbourhood_group
Bronx             1091
Brooklyn         20104
Manhattan        21661
Queens            5666
Staten Island      373
Name: id, dtype: int64
Manhattan is the neighborbood that shows up the most


In [16]:
# Are private rooms the most popular in manhattan?
manhattan = air_bnb[air_bnb['neighbourhood_group'] == 'Manhattan']
manhattan

print(manhattan.groupby('room_type').sum()['number_of_reviews'])
print('Entire home/apt are the most popular stays in Manhattan based on the total number of reviews.')

room_type
Entire home/apt    235147
Private room       209150
Shared room         10272
Name: number_of_reviews, dtype: int64
Entire home/apt are the most popular stays in Manhattan based on the total number of reviews.


In [56]:
# Which hosts are the busiest and based on their reviews?
review_sums = air_bnb.groupby('host_id').sum()['number_of_reviews']
print(review_sums.sort_values(ascending = False))

print(air_bnb[air_bnb['host_id']==37312959])
print(air_bnb[air_bnb['host_id']==344035])
print(air_bnb[air_bnb['host_id']==26432133])
print(air_bnb[air_bnb['host_id']==35524316])
print(air_bnb[air_bnb['host_id']==40176101])

print('The top 5 busiest hosts are: Maya (host_id: 37312959), Brooklyn & Breakfast (host_id: 344035), Danielle(host_id: 26432133), Yasu & Akiko(host_id: 35524316), and Brady (host_id: 40176101)')

host_id
37312959     2273
344035       2205
26432133     2017
35524316     1971
40176101     1818
             ... 
140338526       0
24508767        0
140323391       0
140312311       0
274321313       0
Name: number_of_reviews, Length: 37457, dtype: int64
             id                                    name   host_id host_name  \
9323    7125107  Room steps away from LaGuardia airport  37312959      Maya   
10623   8168619       Steps away from Laguardia airport  37312959      Maya   
13612  10186192     Only Steps away from LaGuardia arpt  37312959      Maya   
14699  11618854  Walking distance  to LaGuardia airport  37312959      Maya   
18045  14133414    Space to rest near LaGuardia Airport  37312959      Maya   

      neighbourhood_group  neighbourhood  latitude  longitude     room_type  \
9323               Queens  East Elmhurst  40.77005  -73.87691  Private room   
10623              Queens  East Elmhurst  40.77006  -73.87683  Private room   
13612              Queens  Ea

In [29]:
# Which neighorhood group has the highest average price?
print(air_bnb.groupby('neighbourhood_group')['price'].mean())

print('Manhattan has the heighest average of $196.88')

neighbourhood_group
Bronx             87.496792
Brooklyn         124.383207
Manhattan        196.875814
Queens            99.517649
Staten Island    114.812332
Name: price, dtype: float64
Manhattan has the heighest average, with a price of $196.88


In [32]:
# Which neighbor hood group has the highest total price?
print(air_bnb.groupby('neighbourhood_group')['price'].sum())

print('Manhattan has the highest total price.')

neighbourhood_group
Bronx              95459
Brooklyn         2500600
Manhattan        4264527
Queens            563867
Staten Island      42825
Name: price, dtype: int64
Manhattan has the highest total price.


In [41]:
#Which top 5 hosts have the highest total price?
highest_total = air_bnb.groupby('host_name').sum().sort_values('price', ascending = False)
print(highest_total.head(5))

print('Top 5 Hosts: Sonder(NYC) (host_id: 71782340547), Blueground (host_id: 24924786136), Michael (host_id: 22673153604), David (host_id: 28633613354), Alex (host_id: 19850092761)')

                       id      host_id     latitude    longitude  price  \
host_name                                                                 
Sonder (NYC)  10885561678  71782340547  13316.25823 -24198.18856  82795   
Blueground     7210036953  24924786136   9451.60418 -17166.13165  70331   
Michael        7430617239  22673153604  16984.89137 -30841.28573  66895   
David          7943862898  28633613354  16414.34392 -29804.53757  65844   
Alex           5496620312  19850092761  11361.10533 -20635.86235  52563   

              minimum_nights  number_of_reviews  reviews_per_month  \
host_name                                                            
Sonder (NYC)            4353               1281             397.56   
Blueground              7470                 29               6.04   
Michael                 4600              11081             475.82   
David                   3754               8103             508.61   
Alex                    1651               6204       

In [57]:
# Who currently has no (zero) availability with a review count of 100 or more?
no_availability = len(air_bnb[(air_bnb['availability_365'] == 0) & (air_bnb['number_of_reviews'] >= 100)])
no_availability

print(f'There are {no_availability} listings with no availability with a review of 100+')

There are 162 listings with no availability with a review of 100+


In [58]:
# What host has the highest total of prices and where are they located??
print(air_bnb.groupby('host_id').sum()[['id','price']].sort_values('price', ascending = False))
print(air_bnb[air_bnb['host_id']==219517861])

print('Sonder has the highest total of prices and is located in the Financial District')

                    id  price
host_id                      
219517861  10885561678  82795
107434423   7210036953  70331
156158778    332529233  37097
205031545   1415225676  35294
30283594    1611854192  33581
...                ...    ...
205820814     27316669     10
180661875     24114389     10
11967922      21869057     10
10132166      20624541      0
13709292      20933849      0

[37457 rows x 2 columns]
             id                                               name    host_id  \
38293  30181691      Sonder | 180 Water | Incredible 2BR + Rooftop  219517861   
38294  30181945         Sonder | 180 Water | Premier 1BR + Rooftop  219517861   
38588  30347708        Sonder | 180 Water | Charming 1BR + Rooftop  219517861   
39769  30937590            Sonder | The Nash | Artsy 1BR + Rooftop  219517861   
39770  30937591        Sonder | The Nash | Lovely Studio + Rooftop  219517861   
...         ...                                                ...        ...   
47691  35871510  

In [59]:
# When did Danielle from Queens last receive a review?
print(air_bnb[(air_bnb['host_name']=='Danielle') & (air_bnb['neighbourhood_group']=='Queens')]['last_review'])

print("Danielle's last review was on 2019-07-08")

7086     2019-07-03
16349           NaN
20403    2019-07-06
21517    2019-07-07
22068    2019-07-06
22469    2019-07-08
27021    2018-01-02
33861    2019-06-20
Name: last_review, dtype: object
Danielle's last review was on 2019-07-08


## Further Questions

1. Which host has the most listings?

In [68]:
most_listings=air_bnb.groupby('host_id', as_index=True)
most_listings[['id']].count().sort_values(['id'], ascending=False).head()

air_bnb[air_bnb['host_id']==219517861]

print('Sonder (host_id: 219517861) had the most listings.')

Sonder (host_id: 219517861) had the most listings.


2. How many listings have completely open availability?

In [72]:
open365 = air_bnb[air_bnb['availability_365'] >= 365]

print(f'There are {open365.id.count()} properties that have completely open availability.')

There are 1295 properties that have completely open availability.


3. What room_types have the highest review numbers?

In [73]:
print(air_bnb.groupby('room_type')['number_of_reviews'].sum())

print('Entire home/apt has the highest number of reviews.')

room_type
Entire home/apt    580403
Private room       538346
Shared room         19256
Name: number_of_reviews, dtype: int64
Entire home/apt has the highest number of reviews.


# Final Conclusion

In this cell, write your final conclusion for each of the questions asked.

Also, if you uncovered some more details that were not asked above, please discribe them here.

-- Add your conclusion --
1. Total Neighborhood Groups: 5. Manhattan is the neighborbood that shows up the most.
2. Entire home/apt are the most popular stays in Manhattan based on the total number of reviews.
3. The top 5 busiest hosts are: Maya (host_id: 37312959), Brooklyn & Breakfast (host_id: 344035), Danielle(host_id: 26432133), Yasu & Akiko(host_id: 35524316), and Brady (host_id: 40176101).
4. Manhattan has the heighest average, with a price of $196.88
5. Manhattan has the highest total price.
6. Top 5 Hosts: Sonder(NYC) (host_id: 71782340547), Blueground (host_id: 24924786136), Michael (host_id: 22673153604), David (host_id: 28633613354), Alex (host_id: 19850092761)
7. There are 162 listings with 0 availability and over 100 reviews.
8. Sonder has the highest total of prices and is located in the Financial District.
9. Danielle's last review was on 2019-07-08

Further Questions:

1. Sonder (host_id: 219517861) had the most listings.
2. There are 1295 properties that have completely open availability.
3. Entire home/apt has the highest number of reviews.
